h2o-algos/src/main/java/hex/tree/gbm/GBMModel.java

```diff
@@ -61,6 +68,66 @@ public GBMModel(Key<GBMModel> selfKey, GBMParameters parms, GBMOutput output) {
     super(selfKey,parms,output);
   }
 
+  @Override
+  public Frame scoreStagedPredictions(Frame frame, Key<Frame> destination_key) {
+    Frame adaptFrm = new Frame(frame);
+    adaptTestForTrain(adaptFrm, true, false);
+
+    final String[] names = makeAllTreeColumnNames();
+    final int outputcols = names.length;
+
+    return new StagedPredictionsTask(this)
+            .doAll(outputcols, Vec.T_NUM, adaptFrm)
+            .outputFrame(destination_key, names, null);
+  }
+
+  private static class StagedPredictionsTask extends MRTask<StagedPredictionsTask> {
+    private final Key<GBMModel> _modelKey;
+
+    private transient GBMModel _model;
+
+    private StagedPredictionsTask(GBMModel model) {
+      _modelKey = model._key;
+    }
+
+    @Override
+    protected void setupLocal() {
+      _model = _modelKey.get();
+      assert _model != null;
+    }
+
+    @Override
+    public void map(Chunk chks[], NewChunk[] nc) {
+      double[] input = new double[chks.length];
+      int contribOffset = _model._output.nclasses() == 1 ? 0 : 1;
+
+      for (int row = 0; row < chks[0]._len; row++) {
+        for (int i = 0; i < chks.length; i++)
+          input[i] = chks[i].atd(row);
+
+        double[] contribs = new double[contribOffset + _model._output.nclasses()];
+        double[] preds = new double[contribs.length];
+
+        int col = 0;
+        for (int tidx = 0; tidx < _model._output._treeKeys.length; tidx++) {
+          Key[] keys = _model._output._treeKeys[tidx];
+          for (int i = 0; i < keys.length; i++) {
+            if (keys[i] != null)
+              contribs[contribOffset + i] += DKV.get(keys[i]).<CompressedTree>get().score(input, _model._output._domains);
```

Comment:
When Nidhi/Megan tried to kill my PR, they will throw 400 to 1000 trees to test it.  Will they frame grow too big in this case if you keep all the predictions from each tree?  Will it be easier to just keep the prediction chosen by the user?  Or, we can let them choose an array of iterations say I want to see the results at the following trees/iterations: [1,10,20,40]?

Reply:
These are good points, it can be added later in a parameter. Users usually create hundreds or thousands of trees - this should be no problem for H2O.

The difference between leaf node assignment and this PR is that this PR deals just with numbers and not strings. That makes it more memory efficient.